src/OJP.Mod

```diff
@@ -388,6 +388,10 @@ MODULE OJP; (*Oberon compiler for JVM in Oberon-07*)
           ELSIF (x.type.form = OJB.Int) & (y.type.form = OJB.Real) THEN
             OJG.RealToInt(y);
             restyp := x.type
+          ELSIF ((x.type.form = OJB.Pointer) & (y.type.form = OJB.Pointer))
+          THEN
+            OJG.PointerToPointer(y, x.type);
```

Comment:
This is a good start. Given the following example:

```
MODULE TestSystemVal;
IMPORT SYSTEM;
  TYPE
    pR0 = POINTER TO R0;
    pR1 = POINTER TO R1;
    R0 = RECORD x: INTEGER END;
    R1 = RECORD(R0) y: INTEGER END;

  VAR
    p0: pR0;
    p1, p: pR1;
BEGIN
  NEW(p1);
  p1.y := 25;
  p0 := p1;
  p := SYSTEM.VAL(pR1, p0);
  WriteInt(p.y); (* 25 *)

  NEW(p1);
  p1.y := 26;
  p := SYSTEM.VAL(pR1, p1);
  WriteInt(p.y) (* 26 *)
END TestSystemVal.
```
Your solution would generate a `CHECKCAST` for both `SYSTEM.VAL` calls.
The equivalent example rewritten using guards in place of `SYSTEM.VAL` will avoid generating the superfluous `CHECKCAST` for `p := SYSTEM.VAL(pR1, p1)`, as we know statically that is always true. 
Try to reuse the implementation of the type guard to solve this problem and avoid introducing `OJG.PointerToPointer`.

Reply:
Bonus points if you add these test cases to `tests/base/TestSystemVal.Mod` :)

Reply:
Thank you for the pointer (no pun intended ;-)). When I started working on this I was trying to see if I could avoid the type guard, and when it became clear that I couldn't I didn't think about using the type guard.

 I will implement all three of your suggestions, but it might be a few days.

Reply:
The new commit calls `OJP.TestType` and then sets `restyp` to `x.type`. I checked the result of `javap` on `TestSystemVal` and found only one `checkcast`. I also added those test cases to `tests/base/TestSystemVal.Mod` and updated `tests/base/TestSystemVal.txt`.